### DISCUSSION TITLE
What happened to yum shell ?

### DISCUSSION THREAD

**Discussion by dletai** [rpm-software-management/dnf5#1923] [Q&A] [answered]
Is there an alternative method to invoke `yum shell` like functionality in dnf5? I was quite surprised to find the `shell` command missing from the dnf5 commands. Is there some discussion I could be referred to, to see why it was dropped/replaced, and what is suggested as an alternative?

**Comment by ppisar** [comment, marked as answer]
There is https://github.com/rpm-software-management/dnf5/issues/1223. YUM shell was actually a Python shell getting access to YUM internals. As DNF5 does not use Python, there was no reasonable way of reimplementing it. While DNF5 team recognizes the need for more a expressive way of specifying the transactions, providing an access to private internals was rejected as unsupportable and unsafe. Currently there is no substitution besides using C++ (or any of the supported language bindings, like Python) API for libdnf5 to write your own application performing the desired operations.

**Comment by mihawk90** [comment]
On an additional note, this is also tracked here:

* #153
* and by extension #429
